### DISCUSSION TITLE
Confused by the way the service works

### DISCUSSION THREAD

**Discussion by saper** [smartlike-org/smartlike#2] [General]
(This is a repost of a [discussion on Mastodon Discourse forum](https://discourse.joinmastodon.org/t/thumb-up-triggered-micro-donations/3978), referring to 29 November 2021 version of a service).

I went to your website but I found it pretty confusing:

- [ ] The login dialog asks me to enter or generate a key. A random password from my random generator has been considered invalid; Maybe the generator should be one clear option to “sign up”.

- [ ] I see some account from mastodon.social already in the system, but could not see if it signed up for anything and how. The author’s guide is kind of confusing, what do I exactly do?

- [ ] I am told to done to someone to begin because “my account is empty”, but I couldn’t figure out how do do that… Tried to click on the “like” then had to login (I though I was logged in already, but maybe it’s a security measure) then I got an empty Smartlike page (or just one with the “Smartlike developer” if I remove “mastodon.social” filter).
    
- [ ] Most obvious place would be to add Smartlike [property to the Mastodon user profile](https://docs.joinmastodon.org/methods/accounts/#retrieve-information), but even I want to post something with the address, what exactly the value of account in `Smartlike: <account>` should be? My public ID looks like a 256 bit value (a SHA-256 hash, maybe), but then I try to donate to you I see that the address looks more like UUID, which is 128 hash.

- [ ] It is not obvious what an “address” is when entering Paypal account data…

- [ ] The links to a “gateway” *(payment gateway?) go currently nowhere for me!

(After some experimenting:)

Wow, I somewhat managed to make a donation and now some money is there but now few strange things continue to happen:

Updating my Identity seems to get positive response (Smartlike network: Profile updated), there is even 200 and “ok” as a response to `POST https://smartlike.org/network`

I have donated to [4855e1d3-ac4a-f6c4-8e03-f66001cef053](https://smartlike.org/channel/4855e1d3-ac4a-f6c4-8e03-f66001cef053) and then some small amount in USD (I chose EUR!) appeared on my account; then it got reduced by one dollar to reach $0.24 and now it ended up at $-0.76 without me doing much! What does it mean?

When I try to “Smartlike new content” I get a “A smokin’ smartlike! You can see how much money you pledged under …” Very funny, but what does it mean? It also disappears so quickly… Hate popups like this!

**Comment by vadim-frolov** [comment]
@saper thank you very much for your feedback! I appreciate it.

> The login dialog asks me to enter or generate a key. A random password from my random generator has been considered invalid; Maybe the generator should be one clear option to “sign up”.

We use BIP39 mnemonic codes that are stored in user password manager or [browser extension](https://chrome.google.com/webstore/detail/smartlike/hbeaghhbggdilbedobkfhneaajmnfipc) (to keep it between page refreshes and donate to current page without visiting smartlike.org, also works with some mobile browsers). The "sign up" should be simplified, true - #3 

> I see some account from mastodon.social already in the system, but could not see if it signed up for anything and how. The author’s guide is kind of confusing, what do I exactly do?

Content item or channel three dot menu|Edit|Claim ownership, follow platform dependent instructions. One can add signature "Smartlike:account" to a content item or channel description. [Example for Mastodon.](https://mastodon.social/web/@vadim_frolov)
Smartliking posts should already work with correct instance resolution. The hosting fee (if set) will go to user instance. Meta data extraction code is collected [here](https://github.com/smartlike-org/meta-extractor).

> I am told to donate to someone to begin because “my account is empty”, but I couldn’t figure out how do do that… Tried to click on the “like” then had to login (I though I was logged in already, but maybe it’s a security measure) then I got an empty Smartlike page (or just one with the “Smartlike developer” if I remove “mastodon.social” filter). 

Will be improved #4 . Thanks.

> Most obvious place would be to add Smartlike property to the Mastodon user profile, but even I want to post something with the address, what exactly the value of account in Smartlike: <account> should be? My public ID looks like a 256 bit value (a SHA-256 hash, maybe), but then I try to donate to you I see that the address looks more like UUID, which is 128 hash.

Account number (UUID) is assigned by the network when funds are added for the first time and account public key is published on the network.

> It is not obvious what an “address” is when entering Paypal account data…

email address that can accept Paypal payments. I added a placeholder hint into the input field.

> The links to a “gateway” *(payment gateway?) go currently nowhere for me!

Fixed. The source code is [here](https://github.com/smartlike-org/smartlike/tree/main/crates/gateway).

>I have donated to 4855e1d3-ac4a-f6c4-8e03-f66001cef053 and then some small amount in USD (I chose EUR!) appeared on my account; then it got reduced by one dollar to reach $0.24 and now it ended up at $-0.76 without me doing much! What does it mean?

Sorry, there was an issue with updating balance displayed in the client. Your balance and transactions themselves look good. I added transaction history to the left-bottom menu under "Log". All user transactions and payouts can be verified there. (The transaction from "09:13:12 29-11" was me adding funds to your account so that you are not blocked while we investigate the issue.)
Payment problem resolution is handled peer-to-peer, there's a messaging tool (Donations|Feedback) and a way to confirm donation receipts manually in case something goes wrong with automatic payment notification (Donations|Inbox|Accept). More detailed instructions are to be added - #5 

> When I try to “Smartlike new content” I get a “A smokin’ smartlike! You can see how much money you pledged under …” Very funny, but what does it mean? It also disappears so quickly… Hate popups like this!

The popup confirms transaction receipt. In user identity settings one can customize the message to show to users who send you donations. Do you have your preferred alternatives? Would they also work for the browser extension? The popup was first introduced there..

**Comment by saper** [comment]
> The popup confirms transaction receipt. In user identity settings one can customize the message to show to users who send you donations. Do you have your preferred alternatives? Would they also work for the browser extension? The popup was first introduced there..

So this means that donation has been successfully sent? If yes, what about something like "You have successfully donated(*) {amount} to {recipient}." Instead of "donated" some consistent verb should be used across the platform (meaning "you have successfully transferred some funds with smartlike"). 

I didn't try/don't want to use the browser extension but in the browser interface maybe the best idea would be to have a space (at the top, maybe) to display recent messages so that they do not disappear that quick.

I am not sure recipients should customize the message. Could be a nice space for creative abuse/pranks/whatever.

**Reply by vadim-frolov** [reply]
Good. We'll prepare more notification alternatives and make a poll to choose a default.
I cannot imagine why a donation recipient would abuse message customization and risk loosing me as their supporter. At the same time customized receipts could turn out to be an important feature for some users.

**Reply by saper** [reply]
> donation recipient would abuse message

For the lulz, you know.

In the meantime I can't check my balance and other stuff since my log in does not seem to work anymore - just filed #7

**Reply by vadim-frolov** [reply]
> For the lulz, you know.

This is acceptable.
We don't try to build a "perfect" service. Our recipe is to add transparent digitally signed micro-transaction processing and decentralized crowdsourced moderation to an existing set of incentives that have proved to work in the real world. With about the same abuse and error rates, the same peer to peer exception handling procedures.

There are donation processors that provide better arbitration. But they typically come with a price: one need to sacrifice privacy and freedom, pay a fee for being controlled, moderated or censored...
